libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/Store.kt

```diff
@@ -77,11 +77,14 @@ open class Store {
             return
         }
 
-        isDispatching = true
         if (!intercept(action)) {
-            reduce(action)
+            isDispatching = true
```

Comment:
Does this all mean this change can be reverted? https://github.com/mapbox/mapbox-navigation-android/pull/6449

Reply:
Yes but as a separate PR. 
Also lets keep the `UIComponent.observe` method improvents.

Reply:
Awesome!